### PR TITLE
added CBS News xml subtitles

### PR DESCRIPTION
**Pull request by iassael**

### CODE REVIEW

File: youtube_dl/extractor/cbsnews.py [hunk 1]
```diff
@@ -85,10 +85,21 @@ def _real_extract(self, url):
                 fmt['ext'] = 'mp4'
             formats.append(fmt)
 
+        cap_url = 'http://www.cbsnews.com/videos/captions/%s.adb_xml' % video_info['mpxRefId']
```
Comment (dstftw) [inline youtube_dl/extractor/cbsnews.py]: Missing `mpxRefId` should not break the extraction.

File: youtube_dl/extractor/cbsnews.py [hunk 2]
```diff
@@ -85,10 +85,21 @@ def _real_extract(self, url):
...
+        
+        subtitles = {
+                        'en': [{
+                            'url': cap_url,
+                            'ext': 'xml'
+
+                        }],
+                    }
```
Comment (dstftw) [inline youtube_dl/extractor/cbsnews.py]: Can you properly format the code? Check with flake8.

**Comment by dstftw** [comment]
A test should be added either in extractor (you can extend existing) or in `test_subtitles.py`.

**Comment by iassael** [comment]
Done

File: youtube_dl/extractor/cbsnews.py [hunk 3]
```diff
@@ -85,10 +100,21 @@ def _real_extract(self, url):
                 fmt['ext'] = 'mp4'
             formats.append(fmt)
 
+        if 'mpxRefId' in video_info:
+            cap_url = 'http://www.cbsnews.com/videos/captions/%s.adb_xml' % video_info['mpxRefId']
+            subtitles = {
+                'en': [{
+                    'url': cap_url,
+                    'ext': 'xml'
```
Comment (yan12125) [inline youtube_dl/extractor/cbsnews.py]: These files are actually [TTML](https://www.w3.org/TR/ttml1/) subtitles.

**Comment by remitamine** [comment]
it's better to extract information from theplatfom url instead of hardcode the subtitle url is here.
i add something simular to https://github.com/rg3/youtube-dl/blob/master/youtube_dl/extractor/cnet.py#L62.

**Comment by remitamine** [comment]
it's almost done but there is a problem in _extract_theplatform_smil it can't handle smil files that contain m3u8 media playlist urls:
```
python2 __main__.py -F 'http://link.theplatform.com/s/dJ5BDC/eSDhwOPqdCjD?format=SMIL&mbr=true'
[ThePlatform] eSDhwOPqdCjD: Downloading SMIL data
[ThePlatform] eSDhwOPqdCjD: Downloading m3u8 information
[ThePlatform] eSDhwOPqdCjD: Downloading m3u8 information
[ThePlatform] eSDhwOPqdCjD: Downloading m3u8 information
[ThePlatform] eSDhwOPqdCjD: Downloading m3u8 information
[ThePlatform] eSDhwOPqdCjD: Downloading m3u8 information
[ThePlatform] eSDhwOPqdCjD: Downloading m3u8 information
[ThePlatform] eSDhwOPqdCjD: Downloading m3u8 information
[ThePlatform] eSDhwOPqdCjD: Downloading JSON metadata
[info] Available formats for eSDhwOPqdCjD:
format code  extension  resolution note
hls-meta-0   mp4        multiple   Quality selection URL 
hls-meta-1   mp4        multiple   Quality selection URL 
hls-meta-2   mp4        multiple   Quality selection URL 
hls-meta-3   mp4        multiple   Quality selection URL 
hls-meta-4   mp4        multiple   Quality selection URL 
hls-meta-5   mp4        multiple   Quality selection URL 
hls-meta-6   mp4        multiple   Quality selection URL 
7            ts         unknown    
...
559          aac        unknown    (best)
```

**Pull request closed without merging**